packages/remix-react/components.tsx

```diff
@@ -516,6 +516,7 @@ export function Meta() {
   let meta: MetaDescriptor[] = [];
   let leafMeta: MetaDescriptor[] | null = null;
   let matches: MetaMatches = [];
+  let error = errors ? Object.values(errors)[0] : null;
```

Comment:
We may also want to test this if 2 different routes throw.  I forget offhand if we can end up with 2 entries in `errors`.  If so we may need this to just walk the matches top down and find the highest error.

Reply:
I'm pretty sure we only take the first error in order of parent > child routes even though the `errors` value is an object by route ids, it'll only ever contain one error. If a parent throws and error we can't go render deeper anyway.

Reply:
https://github.com/remix-run/react-router/blob/main/packages/router/router.ts#L1266-L1268

Reply:
https://github.com/remix-run/react-router/blob/main/packages/router/router.ts#L2050-L2052

Reply:
https://github.com/remix-run/react-router/blob/main/packages/router/router.ts#L2267-L2269

Reply:
There are a couple more but all the same story, please verify if I'm wrong :D

Reply:
ok so we can have 2 errors - but I don't know if we can write a test that causes that since errors are added to the object by the router in top-down order, and key order is preserved in modern JS engines - so this is getting the highest (but it feels risky):

<img width="505" alt="Screenshot 2023-08-08 at 5 52 58 PM" src="https://github.com/remix-run/remix/assets/1609022/558eb951-43fb-439a-8a03-a915f482fa9e">

What do you think about:

```suggestion
  let errorMatch = matches.find((m) => errors && errors[m.id] != null);
  let error = errors && errorMatch ? errors[errorMatch.id] : null;
```

Reply:
It's not obvious - but here's the buried code that can add multiple keys.  If two routes throw and they bubble to different boundaries (i.e., each has their own) - they both go in `errors` for the nearest ancestor boundary route.  But then downstream we only ever render until the highest one and thus never actually leverage the lower one from `errors`.

https://github.com/remix-run/react-router/blob/main/packages/router/router.ts#L3868